sdk/dotnet/Pulumi/Config.cs

```diff
@@ -0,0 +1,202 @@
+        /// Loads an optional configuration value by its key, or <see langword="null"/> if it doesn't exist.
+        /// </summary>
+        public string? Get(string key)
+            => Deployment.InternalInstance.GetConfig(FullKey(key));
+
+        /// <summary>
+        /// Loads an optional configuration value by its key, marking it as a secret, or <see
+        /// langword="null"/> if it doesn't exist.
+        /// </summary>
+        public Output<string>? GetSecret(string key)
+            => MakeClassSecret(Get(key));
+
+        /// <summary>
+        /// Loads an optional configuration value, as a boolean, by its key, or null if it doesn't exist.
+        /// If the configuration value isn't a legal boolean, this function will throw an error.
+        /// </summary>
+        public bool? GetBoolean(string key)
+        {
+            var v = Get(key);
+            return v == null ? default(bool?) :
+                   v == "true" ? true :
+                   v == "false" ? false : throw new ConfigTypeException(FullKey(key), v, nameof(Boolean));
+        }
+
+        /// <summary>
+        /// Loads an optional configuration value, as a boolean, by its key, making it as a secret or
+        /// null if it doesn't exist. If the configuration value isn't a legal boolean, this
+        /// function will throw an error.
+        /// </summary>
+        public Output<bool>? GetSecretBoolean(string key)
+            => MakeStructSecret(GetBoolean(key));
+
+        /// <summary>
+        /// Loads an optional configuration value, as a number, by its key, or null if it doesn't exist.
+        /// If the configuration value isn't a legal number, this function will throw an error.
+        /// </summary>
+        public int? GetInt32(string key)
+        {
+            var v = Get(key);
+            return v == null
+                ? default(int?)
+                : int.TryParse(v, out var result)
+                    ? result
+                    : throw new ConfigTypeException(FullKey(key), v, nameof(Int32));
+        }
+
+        /// <summary>
+        /// Loads an optional configuration value, as a number, by its key, marking it as a secret
+        /// or null if it doesn't exist.
+        /// If the configuration value isn't a legal number, this function will throw an error.
+        /// </summary>
+        public Output<int>? GetSecretInt32(string key)
+            => MakeStructSecret(GetInt32(key));
+
+        /// <summary>
+        /// Loads an optional configuration value, as an object, by its key, or null if it doesn't
+        /// exist. This works by taking the value associated with <paramref name="key"/> and passing it
+        /// to <see cref="JsonConvert.DeserializeObject{T}(string)"/>.
+        /// </summary>
+        public T? GetObject<T>(string key) where T : class
+        {
+            var v = Get(key);
+            try
+            {
+                return v == null ? null : JsonConvert.DeserializeObject<T>(v);
```

Comment:
Any reason we can't still use the built-in Json code? `System.Text.Json.JsonSerializer.Deserialize<T>(string)`?

https://docs.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer?view=netcore-3.0

Reply:
oh!  i wasn't aware of that one.  Thanks!